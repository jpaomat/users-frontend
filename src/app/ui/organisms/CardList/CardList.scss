@import '../../../../assets/scss';
.user-list {
    width: 90%;
    margin: 100px auto;
    border-radius: 10px;
    margin-bottom: 4px;
    position: relative;
    background-color: white;
}

@media screen and (min-width: $break-desktop) {
    .user-list {
        width: 70%;
        margin: 10px auto;
    }
}