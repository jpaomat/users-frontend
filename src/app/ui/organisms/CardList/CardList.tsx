import { CardItemProps, CardListProps } from '../../../interfaces';
import { CardItem } from '../../molecules';
import './CardList.scss';

export const CardList = ({ dataList }: CardListProps) => {
	
	return (
		<div className='user-list'>
			<ul className='list-group'>
				{dataList.map((dataItem: CardItemProps, id) => (
					<CardItem key={id} {...dataItem} />
				))}
			</ul>
		</div>
	);
};
