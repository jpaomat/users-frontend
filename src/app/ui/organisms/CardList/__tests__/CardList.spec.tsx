import { render, screen } from '@testing-library/react';
import { defaultProps } from './mocks';
import { CardList } from '../CardList';

describe('<UserItem /> component', () => {

	beforeEach(()=> jest.clearAllMocks);
	
	test('should render component', () => {
		const { container } = render(
			<CardList dataList={defaultProps}/>
		);
		expect(container).toMatchSnapshot();
	});

	it('should find the description text in the component', () => {
		render(<CardList dataList={defaultProps}/>);
		expect(screen.getByText(defaultProps[0].description)).toBeTruthy();
	});
});