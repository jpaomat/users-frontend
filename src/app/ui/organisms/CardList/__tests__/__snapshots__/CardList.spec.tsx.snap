// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> component should render component 1`] = `
<div>
  <div
    class="user-list"
  >
    <ul
      class="list-group"
    >
      <li
        class="list-group-item card-item"
      >
        <div
          class="card-item__content"
        >
          <div>
            <blockquote
              class="blockquote"
            >
              <p
                class="font-16"
              >
                Description item
              </p>
            </blockquote>
            <figcaption
              class="blockquote-footer"
            >
              <cite
                title="Source Title"
              />
            </figcaption>
          </div>
          <div
            class="item-info__aditional"
          >
            <p
              class="text-primary font-14"
            >
              test
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
